bokehjs/test/unit/core/util/svg.ts

```diff
@@ -0,0 +1,108 @@
+    const size = {width: 150, height: 200}
+    const ctx = new SVGRenderingContext2D(size)
+
+    const test = (_ctx: SVGRenderingContext2D | CanvasRenderingContext2D) => {
+      // Draw shapes
+      for (let i = 0; i <= 3; i++) {
+        for (let j = 0; j <= 2; j++) {
+          _ctx.beginPath()
+          const x             = 25 + j * 50                 // x coordinate
+          const y             = 25 + i * 50                 // y coordinate
+          const radius        = 20                          // Arc radius
+          const startAngle    = 0                           // Starting point on circle
+          const endAngle      = Math.PI + (Math.PI * j) / 2 // End point on circle
+          const counterclockwise = i % 2 == 1               // Draw counterclockwise
+
+          _ctx.arc(x, y, radius, startAngle, endAngle, counterclockwise)
+
+          if (i > 1) {
+            _ctx.fill()
+          } else {
+            _ctx.stroke()
+          }
+        }
+      }
+    }
+
+    test(ctx)
+    const svg = ctx.get_serialized_svg()
+    compare_on_dom(test, svg, size)
+
+    expect(svg).to.be.equal(`<svg version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="150" height="200"><defs/><path fill="none" stroke="#000000" paint-order="fill" d="M 45 25 A 20 20 0 0 1 5 25.000000000000004" stroke-miterlimit="10" stroke-dasharray=""/><path fill="none" stroke="#000000" paint-order="fill" d="M 95 25 A 20 20 0 1 1 75 5" stroke-miterlimit="10" stroke-dasharray=""/><path fill="none" stroke="#000000" paint-order="fill" d="M 145 25 A 20 20 0 1 1 144.99999000000082 24.98000000333332" stroke-miterlimit="10" stroke-dasharray=""/><path fill="none" stroke="#000000" paint-order="fill" d="M 45 75 A 20 20 0 1 0 5 75" stroke-miterlimit="10" stroke-dasharray=""/><path fill="none" stroke="#000000" paint-order="fill" d="M 95 75 A 20 20 0 0 0 75 55" stroke-miterlimit="10" stroke-dasharray=""/><path fill="none" stroke="#000000" paint-order="fill" d="M 145 75 A 20 20 0 1 0 144.99999000000082 75.01999999666667" stroke-miterlimit="10" stroke-dasharray=""/><path fill="#000000" stroke="none" paint-order="stroke" d="M 45 125 A 20 20 0 0 1 5 125"/><path fill="#000000" stroke="none" paint-order="stroke" d="M 95 125 A 20 20 0 1 1 75 105"/><path fill="#000000" stroke="none" paint-order="stroke" d="M 145 125 A 20 20 0 1 1 144.99999000000082 124.98000000333332"/><path fill="#000000" stroke="none" paint-order="stroke" d="M 45 175 A 20 20 0 1 0 5 175"/><path fill="#000000" stroke="none" paint-order="stroke" d="M 95 175 A 20 20 0 0 0 75 155"/><path fill="#000000" stroke="none" paint-order="stroke" d="M 145 175 A 20 20 0 1 0 144.99999000000082 175.01999999666668"/></svg>`)
+  })
+
+
+  it("should fill a red rectangle correctly", () => {
+    const size = {width: 50, height: 50}
+    const ctx = new SVGRenderingContext2D(size)
+
+    const test = (_ctx: SVGRenderingContext2D | CanvasRenderingContext2D) => {
```

Comment:
Leading `_` is an indicator of an unused argument. Preferably rename this to `ctx`. Given that we use lexical scoping exclusively, then there is no room for confusion between both contexts. However, you could also shuffle this code around, as `size` and `ctx` are not used in `test`, so they could be defined at later, further clarifying the code.

Reply:
As long as public API testing goes, `SVGRenderingContext2D` and `CanvasRenderingContext2D` should provide exactly (or almost exactly given my uncertainty about `drawImage()`). I'm guessing there may be a problem with svg context not implementing canvas context, though that should be fixed on its own. Currently svg context is defined as:
```ts
export class SVGRenderingContext2D /*implements CanvasRenderingContext2D*/ { ... }
```
but it should work with `implements` clause uncommented.